tests/drivers/i2c/i2c_slave_api/src/main.c

```diff
@@ -202,31 +187,43 @@ void test_eeprom_slave(void)
 
 	LOG_INF("EEPROM %s Attached !", label_eeprom1);
 
+	/* Figure out which slave belongs to which EEPROM */
```

Comment:
There's something really fishy here.  While the association of an EEPROM instance with a devicetree node can theoretically change between builds, this test is standalone and the devices shouldn't need disambiguation.

When I did this in https://github.com/pabigot/zephyr/commits/pr/26003 the test passes on native_posix (that solution would require adding the node labels to the other overlays as well).  But there's something else funny going on here.  Too much changed in this test for me to figure out what led you to conclude this sort of detection was necessary.  It really shouldn't be.

Reply:
BTW: "this" in the comment above was referencing a paragraph about using the devicetree nodelabel to access a specific EEPROM rather than assuming the instance number was meaningful, which is demonstrated in the commit in my repo.  There was a bunch of commentary about that either in the review summary or the comment that disappeared somehow, including a note that if this path is taken the other devicetree overlays need to be updated to add the node labels.

Reply:
Each I2C bus has its own AT24 chip. Yes the AT24s use the same driver but they are at different I2C addresses. From my reading I don't think I should rely on the ordering of these devices as referenced by DT_INST() ?

Reply:
That's correct: in a real application you'd be getting them based on the node label (`eeprom_1` and `eeprom_2` in the example I provided).  Doing that ensures you know which one you're getting.

However, in a single program like this where there's no persistance between runs there should be no reason to have to disambiguate the devices: the driver for each doesn't change while the code is running, and the test doesn't care which one was which as long as you consistently use the right address with the right bus controller for each one.

Reply:
OK. Well I can make it work like it was so I've dropped this one.

The thing that bugged me is that we are setting the contents of each eeprom through one method (a 'direct' call to eeprom_slave_program()) and then reading it via another (using a I2C bus and address). So I wasn't sure that would be a stable mapping. I guess you have confirmed that it is, or that I misunderstand what is going on.

Reply:
As long as the bus and address are obtained from the devicetree node that's used to identify the specific EEPROM yes it's stable.  It's more clear when the node is identified with an explicit name like:

    #define NODE_EP0 DT_INST(0, atmel_at24)
    #define NODE_EP1 DT_INST(1, atmel_at24)

Reply:
Hmm well I still need to swap it from STM32. So perhaps you can tell me what the solution is...I've pushed the latest code

Reply:
What do you mean you still need to swap it from STM32?  Are you running on hardware with two eeproms?  I see two custom overlays for STM32 boards for the test, one of which mentions an external RF EEPROM connector, though the corresponding daughterboard is apparently obsolete.

How can I reproduce what you see?

Reply:
I mean that posix needs the devices to be around the other way.

This is what I am using:

 west build -f -b native_posix tests/drivers/i2c/i2c_slave_api/ -d posix 
./posix/zephyr/zephyr.elf